tools/gui/qtcodegen.py

```diff
@@ -0,0 +1,335 @@
+                        'Beispiel: 02.03.1982\n')
+                    if ok:
+                        geburtsdatum = str(text)
+                        validate_datum(geburtsdatum)
+                        self.worker.signalUpdateData.emit("GEBURTSDATUM",geburtsdatum)
+                        break
+                    else:
+                        self.hardClose()
+                        break
+                except ValidationError as exc:
+                    QtWidgets.QMessageBox.critical(self, "Geburtsdatum ungültiges Format", "Das Datum entspricht nicht dem richtigen Format (DD.MM.YYYY).")
+                    
+        elif dlgType == "SMSCODE":
+            text, ok = QtWidgets.QInputDialog.getText(self, 'SMS Code', 
+                'Du erhältst gleich eine SMS mit einem Code zur Bestätigung deiner Telefonnummer\n'
+                'Bitte trage nachfolgend den SMS-Code ein.\n'
+                'Beispiel: 551-550\n')
+            if ok:
+                sms_pin = str(text).replace("-", "")
+                self.worker.signalUpdateData.emit("SMSCODE",sms_pin) 
+            else:
+                self.hardClose()
+        elif dlgType == "SMSCODE_OK":
+            QtWidgets.QMessageBox.information(self, "Erfolgreich", "Code erfolgreich generiert. Du kannst jetzt mit der Terminsuche fortfahren.")
+            self.worker.signalUpdateData.emit("SMSCODE_OK","") 
+            
+    def showDlg(self, strMode, strTxt):
+        if strMode == "MISSING_KONTAKT":
+            ret = QtWidgets.QMessageBox.critical(self, "Kontaktdaten ungültig",
+                "Die Kontakdaten sind nicht korrekt!.\n\nBitte Datei neu erstellen!", QMessageBox.StandardButton.Ok)
+            if ret == QMessageBox.StandardButton.Ok:
+                self.hardClose()
+        elif strMode == "CRITICAL_CLOSE":
+            ret = QtWidgets.QMessageBox.critical(self, "Error", strTxt, QMessageBox.StandardButton.Ok)
+            if ret == QMessageBox.StandardButton.Ok:
+                self.hardClose()
+
+                
+    # force to close the dialog without confirmation
+    def hardClose(self):
+        self._hardClose = True
+        self.close()
+         
+    def closeEvent(self, event):
+        """
+        Wird aufgerufen, wenn die Anwendung geschlossen wird
+        """
+
+        if self.thread.isRunning():
+            if self._hardClose is False:
+                res = QtWidgets.QMessageBox.warning(self, "Suche beenden", "Suche wirklich beenden?\n",
+                                                    (QMessageBox.StandardButton.Ok | QMessageBox.StandardButton.Cancel))
+
+                if res != QMessageBox.StandardButton.Ok:
+                    event.ignore()
+                    return
+                
+            
+        #exit
+        if self.parent is not None:
+            self.parent.enableCodeBtn.emit() # enable Code Btn again
```

Comment:
Das Signal gibts nicht mehr in gui.py / Der Button wird ja nicht mehr deaktiviert
```suggestion

```